File: .claude/worktrees/focused-mclaren-b8edd5/scripts/generate-sitemap.js
/**
 * generate-sitemap.js
 * Reads index.html and extracts all presentation/microsite URLs,
 * then writes sitemap.xml to the project root.
 *
 * Usage:  node scripts/generate-sitemap.js
 * Or add to package.json scripts: "sitemap": "node scripts/generate-sitemap.js"
 *
 * UPDATE BASE_URL when you move to Vercel or a custom domain.
 */

const fs = require('fs');
const path = require('path');

const BASE_URL = 'https://chukwumaonyeije.github.io/mfm-presentations';
const ROOT = path.join(__dirname, '..');
const indexHtml = fs.readFileSync(path.join(ROOT, 'index.html'), 'utf8');

// Extract all href attributes pointing to decks/ or microsites/ HTML files
const hrefRegex = /href="((decks|microsites)\/[^"]+\.html)"/g;
const urls = new Set();
urls.add(''); // root URL

let match;
while ((match = hrefRegex.exec(indexHtml)) !== null) {
  // Skip external URLs (descript.com videos, etc.) and infographic images
  const href = match[1];
  if (!href.startsWith('http') && href.endsWith('.html')) {
    urls.add('/' + href);
  }
}

const today = new Date().toISOString().split('T')[0];

const entries = Array.from(urls)
  .sort()
  .map(url => {
    const isRoot = url === '';
    const priority = isRoot ? '1.0' : url.includes('microsites') ? '0.9' : '0.8';
    return `  <url>
    <loc>${BASE_URL}${isRoot ? '/' : url}</loc>
    <lastmod>${today}</lastmod>
    <changefreq>monthly</changefreq>
    <priority>${priority}</priority>
  </url>`;
  })
  .join('\n');

const sitemap = `<?xml version="1.0" encoding="UTF-8"?>
<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
${entries}
</urlset>
`;

fs.writeFileSync(path.join(ROOT, 'sitemap.xml'), sitemap);
console.log(`✅  sitemap.xml generated with ${urls.size} URLs`);
console.log(`    Base URL: ${BASE_URL}`);
console.log(`    Tip: Update BASE_URL in this script when you move to Vercel/custom domain.`);

File: generate_sitemap.py
import json
import os
from datetime import datetime

BASE_URL = "https://openmfm.org"

with open('landing-page/data/presentations.json', 'r', encoding='utf-8') as f:
    data = json.load(f)

presentations = data['presentations']
microsites = data['microsites']

urls = []

# Main pages
urls.append(('/', '1.0'))
urls.append(('/library', '0.9'))

# Presentations
for p in presentations:
    urls.append((f"/{p['href']}", '0.8'))

# Microsites (relative URLs only)
for m in microsites:
    if not m['href'].startswith('http'):
        urls.append((f"/{m['href']}", '0.7'))

today = datetime.now().strftime('%Y-%m-%d')

xml_lines = [
    '<?xml version="1.0" encoding="UTF-8"?>',
    '<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">'
]

for url, priority in urls:
    xml_lines.append('  <url>')
    xml_lines.append(f'    <loc>{BASE_URL}{url}</loc>')
    xml_lines.append(f'    <lastmod>{today}</lastmod>')
    xml_lines.append(f'    <changefreq>monthly</changefreq>')
    xml_lines.append(f'    <priority>{priority}</priority>')
    xml_lines.append('  </url>')

xml_lines.append('</urlset>')

xml_content = '\n'.join(xml_lines)

os.makedirs('landing-page/public', exist_ok=True)
output_path = 'landing-page/public/sitemap.xml'
with open(output_path, 'w', encoding='utf-8', newline='\n') as f:
    f.write(xml_content)
print(f"Generated {output_path} with {len(urls)} URLs.")

# Also keep root sitemap.xml in sync (canonical openmfm.org URLs)
root_output_path = 'sitemap.xml'
with open(root_output_path, 'w', encoding='utf-8', newline='\n') as f:
    f.write(xml_content)
print(f"Generated {root_output_path} with {len(urls)} URLs.")
